custom_components/govee_ble_hci/govee_advertisement.py

```diff
@@ -107,16 +107,17 @@ def __init__(self, data: bytes):
                 # Negative temperature stored an two's complement
                 temp_lsb_int = int(temp_lsb, 16)
                 self.temperature = float(twos_complement(temp_lsb_int) / 100)
-                self.battery = int(self.mfg_data[7])
+                if len(self.mfg_data) == 9:
+                    self.battery = int(self.mfg_data[7])
         except (ValueError, IndexError):
             pass
 
     def check_is_gvh5074(self) -> bool:
-        """Check if mfg data is that of Govee H5074, 5072 or the new 5102"""
-        return self._mfg_data_check(9, 6)
+        """Check if mfg data is that of Govee H5074 or generic Govee"""
+        return self._mfg_data_check(9, 6) or self._mfg_data_check(11, 6)
```

Comment:
Instead of piggybacking off the `check_is_gvh5074` create `check_is_gvh5051`

Reply:
Easy enough, but the 5074 also has these messages and as of right now they are just being ignored.  Do we care about processing those messages or is it safe to just ignore them?

Reply:
What I mean is leave the check and processing of the gv5074 as is and add a the necessary check and processing for gvh5051.  I am ok with some code redundancy for the time being if it makes it clearer to understand how each device is handled.  It would be something similar to this: 

```python
            .....
            elif self.check_is_gvh5074():
                mfg_data_5074 = hex_string(self.mfg_data[3:7]).replace(" ", "")
                temp_lsb = mfg_data_5074[2:4] + mfg_data_5074[0:2]
                hum_lsb = mfg_data_5074[6:8] + mfg_data_5074[4:6]
                self.packet = temp_lsb + hum_lsb
                self.humidity = float(int(hum_lsb, 16) / 100)
                # Negative temperature stored an two's complement
                temp_lsb_int = int(temp_lsb, 16)
                self.temperature = float(twos_complement(temp_lsb_int) / 100)
                self.battery = int(self.mfg_data[7])
           elif self.check_is_gvh5051():
                mfg_data_5051 = hex_string(self.mfg_data[3:7]).replace(" ", "")
                temp_lsb = mfg_data_5051[2:4] + mfg_data_5051[0:2]
                hum_lsb = mfg_data_5051[6:8] + mfg_data_5051[4:6]
                self.packet = temp_lsb + hum_lsb
                self.humidity = float(int(hum_lsb, 16) / 100)
                # Negative temperature stored an two's complement
                temp_lsb_int = int(temp_lsb, 16)
                self.temperature = float(twos_complement(temp_lsb_int) / 100)
                self.battery = 0
          .....

    def check_is_gvh5051(self) -> bool:
        """Check if mfg data is that of Govee H5051."""
        return self._mfg_data_check(11, 6)
```

Reply:
Can you also make a PR with only these changes and not the ones from #16 ?